images/router/haproxy/conf/haproxy-config.template

```diff
@@ -226,6 +226,11 @@ backend be_edge_http_{{$cfgIdx}}
     {{ else }}
   balance leastconn
     {{ end }}
+    {{ with $value := index $cfg.Annotations "router.openshift.io/haproxy.timeout"}}
+      {{if (matchString "[0-9]*(us|ms|s|m|h|d)?" $value) }}
```

Comment:
Should we make anchoring the front and back the default in matchString?  i.e. wrap the match they give us with ^(...)$

If we want to match a substr, then we can say .*thing.*

It feels like we will bite people with this behavior.